convert2rhel/systeminfo.py

```diff
@@ -152,49 +155,74 @@ def get_system_release_file_content():
 
         return redhatrelease.get_system_release_content()
 
-    def _get_system_name(self, system_release_content=None):
-        content = self.system_release_file_content if not system_release_content else system_release_content
-        name = re.search(r"(.+?)\s?(?:release\s?)?\d", content).group(1)
-        return name
+    def parse_system_release_content(self, system_release_content=None):
+        """Parse the content of the system release string
 
-    def _get_system_version(self, system_release_content=None):
-        """Return a namedtuple with major and minor elements, both of an int type.
+        If system_release_content is not provided we use the content from the self.system_release_file_content.
 
-        Examples:
-        Oracle Linux Server release 7.8
-        CentOS Linux release 7.6.1810 (Core)
-        CentOS Linux release 8.1.1911 (Core)
+        :param system_release_content: The contents of the system_release file if needed.
+        :type system_release_content: str
+        :returns: A dictionary containing the system release information
+        :rtype: dict[str, str]
         """
+
         content = self.system_release_file_content if not system_release_content else system_release_content
```

Comment:
This is mainly used for easier unit testing, right? I was wondering if it's needed, but I see this usecase

Reply:
Yes, I think it is not actively used, but I wanted to keep it, as it simplifies testing and generally allows to tweak the perception of a system-release without editing the file on the filesystem.

Could be useful in scenarios when you want to keep the system read-only.